app/models/piece.rb

```diff
@@ -1,3 +1,64 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
+
+  def check_square(x, y)
```

Comment:
Here is more info on that. http://pragmati.st/2012/03/24/the-elements-of-ruby-style-predicate-methods/ :)